src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2346,12 +2346,14 @@ public override BoundNode VisitNullCoalescingAssignmentOperator(BoundNullCoalesc
             BoundExpression rightOperand = node.RightOperand;
             int leftSlot = MakeSlot(leftOperand);
 
-            // The assignment to the left below needs the declared type from VisitLvalue, but the hidden
-            // unnecessary check diagnostic needs the current adjusted type of the slot
             TypeWithAnnotations targetType = VisitLvalueWithAnnotations(leftOperand);
             var leftState = this.State.Clone();
             LearnFromNonNullTest(leftOperand, ref leftState);
             LearnFromNullTest(leftOperand, ref this.State);
+            if (node.IsNullableValueTypeAssignment)
+            {
+                targetType = TypeWithAnnotations.Create(node.Type, NullableAnnotation.NotAnnotated);
```

Comment:
>targetType = TypeWithAnnotations.Create(node.Type, NullableAnnotation.NotAnnotated); [](start = 16, length = 84)

Can we add a nullability test to cover this? (`(nullableInt ??= default).ToString()` or something along those lines)